packages/proto-loader/src/index.ts

```diff
@@ -115,6 +115,53 @@ export interface EnumTypeDefinition extends ProtobufTypeDefinition {
   format: 'Protocol Buffer 3 EnumDescriptorProto';
 }
 
+export enum IdempotencyLevel {
+  IDEMPOTENCY_UNKNOWN = 0,
+  NO_SIDE_EFFECTS = 1,
+  IDEMPOTENT = 2
+}
+
+export interface NamePart {
+  namePart: string;
+  isExtension: boolean;
+}
+
+export interface UninterpretedOption {
+  name?: (NamePart[]|null);
+  identifierValue?: (string|null);
+  positiveIntValue?: (number|Long|string|null);
+  negativeIntValue?: (number|Long|string|null);
```

Comment:
I don't like the variance in these field types. I assume it comes from the options that were passed when loading the proto files. If possible, it would be better to have a specific type for this, if possible.

Reply:
Could we leave custom extension type as `unknown`?
```ts
export interface MethodOptions {
  deprecated?: boolean;
  idempotency_level?: IdempotencyLevel|keyof typeof IdempotencyLevel;
  uninterpreted_option?: UninterpretedOption[];
  [k: string]: unknown
}
```

And clean up other field types too, do you still need separated type for `UninterpretedOption` values?
```ts
export enum IdempotencyLevel {
  IDEMPOTENCY_UNKNOWN = 0,
  NO_SIDE_EFFECTS = 1,
  IDEMPOTENT = 2
}

export interface NamePart {
  namePart: string;
  isExtension: boolean;
}

export interface UninterpretedOption {
  name?: NamePart[];
  identifierValue?: string;
  positiveIntValue?: number|Long|string;
  negativeIntValue?: number|Long|string;
  doubleValue?: number|null;
  stringValue?: Uint8Array|string;
  aggregateValue?: string;
}

export interface CustomHttpPattern {
  kind?: string;
  path?: string;
}
```

Reply:
Custom extension types as `unknown` seems like a good solution for that part to me. Removing the `null` values from the other fields is good too, but I'm more concerned about the `number|Long|string` part and similar. It would be better if we could declare in advance which specific type each field will have, either by fixing the corresponding protobuf options in advance or by having type conversion code on top of that. This also applies to `stringValue` and to `idempotency_level`.

Also, I just realized, what is the deal with the inconsistent field name format between `MethodOptions` and the other interfaces? In `descriptor.proto`, they all use snake_case.

Reply:
Hi, sorry for this late response.

I updated & cleaned up `MethodOptions` and other related interfaces. As `protobufjs` won't handle 64-bit integers in method options, it will output just `number` at the moment.

I also updated new test cases; please check and let me know if you find any other issues.